portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,32 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> messages;
```

Comment:
Use interface (sorry if this is a redundant comment - I think this might be one of those changes where your previous change is included).

Reply:
Done.